src/ProxyClient/Varnish.php

```diff
@@ -61,18 +63,35 @@ class Varnish extends HttpProxyClient implements BanCapable, PurgeCapable, Refre
      */
     public function invalidateTags(array $tags)
     {
+        $tagMode = $this->options['tag_mode'];
+        $invalidationMethod = "{$tagMode}TagChunk";
         $escapedTags = array_map('preg_quote', $this->escapeTags($tags));
 
-        $chunkSize = $this->determineTagsPerHeader($escapedTags, '|');
+        $chunkSize = $this->determineTagsPerHeader($escapedTags, 'ban' === $tagMode ? '|' : ' ');
 
         foreach (array_chunk($escapedTags, $chunkSize) as $tagchunk) {
-            $tagExpression = sprintf('(^|,)(%s)(,|$)', implode('|', $tagchunk));
-            $this->ban([$this->options['tags_header'] => $tagExpression]);
+            $this->$invalidationMethod($tagchunk);
```

Comment:
i would prefer to be explicit here and say `if ('ban' === $tagMode) { $this->banTags(); } else {$this->purgekeys();}`  or something like that.

Reply:
Ok, tought you said you did not want condition.

Reply:
sorry, not sure what i meant anymore. but looking at the code i think a condition is more readable.

Reply:
I agree, just way beyond caring to argue anymore with the input.